test/framework/gardener_utils.go

```diff
@@ -495,6 +498,10 @@ func setHibernation(shoot *gardencorev1beta1.Shoot, hibernated bool) {
 // VerifyNoRunningPods verifies that no control plane pods are running for a given shoot.
 // If any control plane pods are found to be running, returns an error with their names. Otherwise, returns nil.
 func (f *GardenerFramework) VerifyNoRunningPods(ctx context.Context, shoot *gardencorev1beta1.Shoot) error {
+	if v1beta1helper.IsWorkerless(shoot) {
+		return nil
+	}
+
```

Comment:
Same like above - better prevent the caller from calling it for workerless shoots